### PR TITLE
Allow for analysis under rebar3_hank

### PR DESCRIPTION
**Pull request by paulo-ferraz-oliveira**
(also move to most recent rebar3_lint)

### CODE REVIEW

File: elvis.config [hunk 1]
```diff
@@ -7,11 +7,6 @@
         filter => "*.erl",
         ruleset => erl_files
        },
-      #{dirs => ["."],
```
Comment (paulo-ferraz-oliveira) [inline elvis.config:10]: `rebar3_lint` (now the most recent version) told me this wasn't being used... `rebar3_lint` is also killing it with fire! 😄

File: rebar.config [hunk 2]
```diff
@@ -96,4 +96,18 @@
 
 %% == Shell ==
 
-{plugins, [rebar3_auto, {rebar3_lint, "0.2.0"}, rebar3_hex]}.
+{project_plugins, [
+    rebar3_auto,
+    rebar3_lint,
```
Comment (paulo-ferraz-oliveira) [inline rebar.config:101]: Moved to the most recent version. Hope that's Ok. If not, I can revert.

File: src/cowboy_swagger.erl [hunk 3]
```diff
@@ -70,8 +72,8 @@
 -type metadata() :: trails:metadata(swagger_map()).
 -export_type([swagger_map/0, metadata/0]).
 
--type swagger_version() :: swagger_2_0
-                         | openapi_3_0_0.
+-type swagger_version() :: 'swagger_2_0'
```
Comment (paulo-ferraz-oliveira) [inline src/cowboy_swagger.erl:75]: All these `'` changes are due to `rebar3_lint`'s most recent version.

File: src/cowboy_swagger.erl [hunk 4]
```diff
@@ -11,6 +11,8 @@
 -export([filter_cowboy_swagger_handler/1]).
 -export([get_existing_definitions/1]).
 
+-hank([{unnecessary_function_arguments, [is_visible/2]}]).
```
Comment (elbrujohalcon) [inline src/cowboy_swagger.erl:14]: A comment above this line explaining why it's needed would be a nice addition.

File: src/cowboy_swagger.erl (hunk 3 above)
Comment (elbrujohalcon) [reply src/cowboy_swagger.erl:75]: Why don't we change the `regex` in `elvis.config`, instead of adding `'`s?

Comment (elbrujohalcon) [reply src/cowboy_swagger.erl:75]: …or adding an `-elvis(…).` attribute here.

**Review by elbrujohalcon — CHANGES_REQUESTED** [review]

File: src/cowboy_swagger.erl (hunk 3 above)
Comment (paulo-ferraz-oliveira) [reply src/cowboy_swagger.erl:75]: I personally like the regex as-is, but you might want to be more permissive, sure. Let me know what you prefer.

File: src/cowboy_swagger.erl (hunk 4 above)
Comment (paulo-ferraz-oliveira) [reply src/cowboy_swagger.erl:14]: Sure. I'll do it later.

File: src/cowboy_swagger.erl (hunk 3 above)
Comment (elbrujohalcon) [reply src/cowboy_swagger.erl:75]: The regex is good _in general_, but it doesn't apply to this particular module. That's fine. We can adjust it for this module. That's what `-elvis(…).` is for, isn't it?

Comment (paulo-ferraz-oliveira) [reply src/cowboy_swagger.erl:75]: 😄

Comment (paulo-ferraz-oliveira) [reply src/cowboy_swagger.erl:75]: This occurs in two modules, actually, so I'm going with the "`elvis.config`-based exception" approach.

File: src/cowboy_swagger.erl (hunk 4 above)
Comment (paulo-ferraz-oliveira) [reply src/cowboy_swagger.erl:15]: Done

```erlang
% is_visible is used as a maps:filter/2 predicate, which requires a /2 arity function
```

File: elvis.config [hunk 5]
```diff
@@ -5,12 +5,8 @@
...
      [#{dirs => ["src", "test"],
         filter => "*.erl",
-        ruleset => erl_files
-       },
-      #{dirs => ["."],
-        filter => "Makefile",
-        rules => [{elvis_project, protocol_for_deps_erlang_mk, #{regex => "(https://.*|[0-9]+([.][0-9]+)*)"}}],
-        ruleset => makefiles
+        ruleset => erl_files,
+        rules => [{elvis_style, atom_naming_convention, #{ regex => "^(([a-z][a-z0-9]*_?)*(_SUITE)?|basePath|swagger_2_0|openapi_3_0_0)$" }}]
```
Comment (elbrujohalcon) [inline elvis.config:9]: HAHAHAHA! You really loved the original regex, my friend! 😝

**Review by elbrujohalcon — APPROVED** [review]

**Comment by elbrujohalcon** [comment]
Do we need to update `katana-test`?

**Comment by paulo-ferraz-oliveira** [comment]
> Do we need to update `katana-test`?

I don't think so. All checks are passing, and I dropped the meta SUITE.

File: rebar.config [hunk 6]
```diff
@@ -24,13 +24,15 @@
 
 {profiles, [
   {test, [
-    {deps, [ {katana_test, "1.0.1"}
-           , {mixer,       "1.1.1", {pkg, inaka_mixer}}
+    {deps, [ {mixer,       "1.1.1", {pkg, inaka_mixer}}
            , {shotgun,     "0.5.0"}
+           , {xref_runner, "1.2.0"}
```
Comment (elbrujohalcon) [inline rebar.config:29]: Why do we need `xref_runner`?

**Review by elbrujohalcon — CHANGES_REQUESTED** [review]

File: rebar.config (hunk 6 above)
Comment (elbrujohalcon) [reply rebar.config:29]: Oh, `xref_SUITE`… That one can also be killed.

Comment (paulo-ferraz-oliveira) [reply rebar.config:29]: Ha, ha, ha, slow laugh, slow clap.

Comment (paulo-ferraz-oliveira) [reply rebar.config]: Removed.

**Review by elbrujohalcon — APPROVED** [review]

**Pull request merged**